src/os/bluestore/BlueStore.cc

```diff
@@ -3506,7 +3506,14 @@ BlueStore::BlobRef BlueStore::ExtentMap::split_blob(
 void BlueStore::Onode::get() {
   if (++nref >= 2 && !pinned) {
     OnodeCacheShard* ocs = c->get_onode_cache();
-    std::lock_guard l(ocs->lock);
```

Comment:
indeed, this looks like one more issue with all this OnodeCaching's stuff.
Could you please follow the "while(true)" approach to select/acquire similar to the one at
void BlueStore::SharedBlob::finish_write(uint64_t seq)
  while (true) {
    BufferCacheShard *cache = coll->cache;
    std::lock_guard l(cache->lock);
    if (coll->cache != cache) {
{
  while (true) {
    BufferCacheShard *cache = coll->cache;
    std::lock_guard l(cache->lock);
    if (coll->cache != cache) {
      dout(20) << __func__
	       << " raced with sb cache update, was " << cache
	       << ", now " << coll->cache << ", retrying"
	       << dendl;
      continue;
    }
    bc._finish_write(cache, seq);
    break;
  }
}

Reply:
yep, that looks like the right approach Igor!